src/evaluation/evaluator.h

```diff
@@ -745,14 +745,17 @@ class Evaluator {
     constexpr movegen::Move
     getBestMove(const BitBoard& board, std::optional<uint8_t> depthInput = std::nullopt)
     {
-        m_startTime
-            = std::chrono::system_clock::now();
-
+        {
+            std::unique_lock<std::mutex> lock(m_timeHandleMutex);
```

Comment:
The sanitizer might flag this as racy but I don't think this solves the issue. 
starttime and endtime are still set outside their own locks. Might be "memory safe" now but it's still racy :)

`getBestMove` is not designed to be thread safe. Why would we ever want to perform multiple searches? :)